pkg/kubelet/kubelet_pods.go

```diff
@@ -228,13 +228,29 @@ func makePortMappings(container *api.Container) (ports []kubecontainer.PortMappi
 	return
 }
 
+// truncatePodHostnameIfNeeded truncates the pod hostname if it's longer than 63 chars.
+func truncatePodHostnameIfNeeded(podName, hostname string) (string, error) {
+	// Cap hostname at 63 chars (specification is 64bytes which is 63 chars and the null terminating char).
+	const hostnameMaxLen = 63
+	if len(hostname) <= hostnameMaxLen {
+		return hostname, nil
+	}
+	truncated := hostname[:hostnameMaxLen]
+	glog.Errorf("hostname for pod:%q was longer than %d. Truncated hostname to :%q", podName, hostnameMaxLen, truncated)
+	// hostname should not end with '-' or '.'
+	truncated = strings.TrimRight(truncated, "-.")
```

Comment:
i'm wondering if we should have put the log statement below this line where we trim off `-.`